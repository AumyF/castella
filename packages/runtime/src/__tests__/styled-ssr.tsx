/**
 * @jest-environment node
 */
import React, { Fragment } from "react";
import { renderToStaticMarkup, renderToString } from "react-dom/server";
import { ServerRenderingCollector, styledComponent } from "..";

describe("styledComponent ssr", () => {
  describe("no ssr", () => {
    it("input", () => {
      const Hello = styledComponent({
        ssr: true,
        css: ".test-foobar{color:red;}",
        className: "test-foobar",
        elementName: "input",
      });

      const str = renderToString(<Hello />);

      expect(str).toMatchSnapshot();
    });
    it("div", () => {
      const Hello = styledComponent({
        ssr: true,
        css: ".test-divdiv{color:red;}",
        className: "test-divdiv",
        elementName: "div",
      });

      const str = renderToString(
        <Hello>
          <p>Hi!</p>
        </Hello>
      );

      expect(str).toMatchSnapshot();
    });
    it("rendered twice", () => {
      const Hello = styledComponent({
        ssr: true,
        css: ".test-divdiv{color:red;}",
        className: "test-divdiv",
        elementName: "div",
      });

      const str = renderToString(
        <Hello>
          <p>Hi!</p>
          <Hello>
            <p>Hoy!</p>
          </Hello>
        </Hello>
      );

      expect(str).toMatchSnapshot();
    });
  });
  describe("with ssr", () => {
    it("input", () => {
      const Hello = styledComponent({
        ssr: true,
        css: ".test-foobar{color:red;}",
        className: "test-foobar",
        elementName: "input",
      });

      const sheet = new ServerRenderingCollector();
      const str = renderToString(sheet.wrap(<Hello />));

      expect(str).toMatchSnapshot();
      expect(
        renderToStaticMarkup(<Fragment>{sheet.getHeadElements()}</Fragment>)
      ).toMatchSnapshot();
    });
    it("div", () => {
      const Hello = styledComponent({
        ssr: true,
        css: ".test-divdiv{color:red;}",
        className: "test-divdiv",
        elementName: "div",
      });

      const sheet = new ServerRenderingCollector();
      const str = renderToString(
        sheet.wrap(
          <Hello>
            <p>Hi!</p>
          </Hello>
        )
      );

      expect(str).toMatchSnapshot();
      expect(
        renderToStaticMarkup(<Fragment>{sheet.getHeadElements()}</Fragment>)
      ).toMatchSnapshot();
    });
    it("rendered twice", () => {
      const Hello = styledComponent({
        ssr: true,
        css: ".test-divdiv{color:red;}",
        className: "test-divdiv",
        elementName: "div",
      });
      const Hello2 = styledComponent({
        ssr: true,
        css: ".test-hello2{color:blue;}",
        className: "test-hello2",
        elementName: "div",
      });

      const sheet = new ServerRenderingCollector();
      const str = renderToString(
        sheet.wrap(
          <Hello>
            <p>Hi!</p>
            <Hello>
              <p>Hoy!</p>
            </Hello>
            <Hello2>wow!</Hello2>
          </Hello>
        )
      );

      expect(str).toMatchSnapshot();
      expect(
        renderToStaticMarkup(<Fragment>{sheet.getHeadElements()}</Fragment>)
      ).toMatchSnapshot();
    });
  });
});
