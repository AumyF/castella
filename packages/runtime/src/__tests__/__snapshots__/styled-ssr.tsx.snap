// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styledComponent ssr no ssr div 1`] = `"<div class=\\"test-divdiv\\" data-reactroot=\\"\\"><p>Hi!</p></div>"`;

exports[`styledComponent ssr no ssr input 1`] = `"<input class=\\"test-foobar\\" data-reactroot=\\"\\"/>"`;

exports[`styledComponent ssr no ssr rendered twice 1`] = `"<div class=\\"test-divdiv\\" data-reactroot=\\"\\"><p>Hi!</p><div class=\\"test-divdiv\\"><p>Hoy!</p></div></div>"`;

exports[`styledComponent ssr with ssr div 1`] = `"<div class=\\"test-divdiv\\"><p>Hi!</p></div>"`;

exports[`styledComponent ssr with ssr div 2`] = `"<style data-castella-styled=\\"test-divdiv\\">.test-divdiv{color:red;}</style>"`;

exports[`styledComponent ssr with ssr input 1`] = `"<input class=\\"test-foobar\\"/>"`;

exports[`styledComponent ssr with ssr input 2`] = `"<style data-castella-styled=\\"test-foobar\\">.test-foobar{color:red;}</style>"`;

exports[`styledComponent ssr with ssr rendered twice 1`] = `"<div class=\\"test-divdiv\\"><p>Hi!</p><div class=\\"test-divdiv\\"><p>Hoy!</p></div><div class=\\"test-hello2\\">wow!</div></div>"`;

exports[`styledComponent ssr with ssr rendered twice 2`] = `"<style data-castella-styled=\\"test-divdiv\\">.test-divdiv{color:red;}</style><style data-castella-styled=\\"test-hello2\\">.test-hello2{color:blue;}</style>"`;
